src/BSON/UTCDateTime.c

```diff
@@ -46,17 +51,43 @@ PHONGO_API zend_class_entry *php_phongo_utcdatetime_ce;
 
 zend_object_handlers php_phongo_handler_utcdatetime;
 
-/* {{{ proto BSON\UTCDateTime UTCDateTime::__construct(integer $milliseconds)
+/* {{{ proto BSON\UTCDateTime UTCDateTime::__construct(integer|DateTime $milliseconds)
    Construct a new UTCDateTime */
 PHP_METHOD(UTCDateTime, __construct)
 {
 	php_phongo_utcdatetime_t    *intern;
-	zend_error_handling       error_handling;
-
+	zend_error_handling          error_handling;
+	zval                        *datetime = NULL;
 
 	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling TSRMLS_CC);
 	intern = Z_UTCDATETIME_OBJ_P(getThis());
 
+	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "|O!", &datetime, php_date_get_date_ce()) == SUCCESS) {
+		int64_t sec, usec;
+
+		if (datetime) {
+			/* Initialize UTCDateTime from DateTime object. The following
+			 * assignments use the same logic as date_format() in php_date.c */
+			php_date_obj *datetime_obj;
+
+			datetime_obj = Z_PHPDATE_P(datetime);
+			sec = datetime_obj->time->sse;
+			usec = (int64_t) floor(datetime_obj->time->f * 1000000 + 0.5);
+		} else {
+			/* Initialize UTCDateTime from current time */
```

Comment:
`zend_parse_parameters()` doesn't have a pattern to pick up `null`, which is equivalent to no argument at all, so this seemed easiest. Otherwise, I would need to change the original `zend_parse_parameters()` calls below (for string/integer parsing) to each allow for `null`.